utils/db.py

```diff
@@ -92,12 +92,14 @@ class Db(Mapping):
         tables, like the mapping interface or :py:meth:`values`.
 
     """
-    def __init__(self, hostname=None, credentials=None):
+    def __init__(self, appliance=None, credentials=None):
         self._table_cache = {}
-        if hostname is None:
+        if appliance is None:
             self.hostname = store.current_appliance.db_address
+            self.db_port = store.current_appliance.db_port
         else:
-            self.hostname = hostname
+            self.hostname = appliance.db_address
+            self.db_port = appliance.db_port
```

Comment:
You can possibly convert `self.db_port` to a property that resolves to a `self.appliance.db_port`, if you know what I mean.

```python
# in __init__
self.appliance = appliance or store.current_appliance

# then

@property
def db_port(self):
    return self.appliance.db_port
```

Same with the hostname.